### PR TITLE
unit-file: support UpheldBy= in [Install] settings (adding Upholds= deps from .upholds/)

### LINKED ISSUE #26896: Support UpheldBy in Install section for symlinks in .upholds directory
### Component

systemd

### Is your feature request related to a problem? Please describe

For systemd units that get loaded in systemd-sysext images after the system is booted we need the `Upholds=` property to ensure that a unit is started. The regular `Install` section properties `WantedBy=`/`RequiredBy=` and their corresponding symlinks to the unit in the `.wants`/`.requires` directories of the specified unit are not enough because they only trigger the start when they are (re)started.
The current solution is to add drop-in unit files that add the `Upholds=` property for, e.g., `multi-user.target`. That's ok but means that the user can't use `systemctl enable`/`systemctl disable` to control the unit and has to use `systemctl unmask`/`systemctl mask` instead.

### Describe the solution you'd like

Allow to specify `UpheldBy=` in the unit's `[Install]` section, and make `systemctl enable`/`systemctl disable` mange the symlink in a `.upholds` directory. The symlinks in this directory should be evaluated to become `Upholds=` properties.

### Describe alternatives you've considered

Leave things as they are and use drop-in units and masking.

### The systemd version you checked that didn't have the feature you are asking for

253

### PR DESCRIPTION
**Pull request by YHNdnzj**
Closes #26896

### CODE REVIEW

File: man/systemctl.xml [hunk 1]
```diff
@@ -367,10 +367,9 @@ Jan 12 10:46:45 example.com bluetoothd[8900]: gatt-time-server: Input/output err
           <listitem>
             <para>Shows units required and wanted by the specified
             units. This recursively lists units following the
-            <varname>Requires=</varname>,
-            <varname>Requisite=</varname>,
-            <varname>ConsistsOf=</varname>,
-            <varname>Wants=</varname>, <varname>BindsTo=</varname>
+            <varname>Requires=</varname>, <varname>Requisite=</varname>,
+            <varname>Wants=</varname>, <varname>ConsistsOf=</varname>,
+            <varname>BindsTo=</varname> and <varname>Upholds=</varname>
```
Comment (yuwata) [inline man/systemctl.xml]: Please use the following rule to add comma.
A, B and/or C -> A, B, and/or C
That is, add a comma before and/or.

File: man/systemd.unit.xml [hunk 2]
```diff
@@ -1895,10 +1900,10 @@
       </tgroup>
     </table>
 
-    <para>Note: <varname>WantedBy=</varname> and <varname>RequiredBy=</varname> are
-    used in the [Install] section to create symlinks in <filename>.wants/</filename>
-    and <filename>.requires/</filename> directories. They cannot be used directly as a
-    unit configuration setting.</para>
+    <para>Note: <varname>WantedBy=</varname>, <varname>RequiredBy=</varname> and <varname>UpheldBy=</varname>
```
Comment (yuwata) [inline man/systemd.unit.xml]: Also here, and below.

**Review by yuwata — CHANGES_REQUESTED** [review]
Could you add several testcases e.g. in test/units/testsuite-23.Upholds.sh.

**Review by yuwata — APPROVED** [review]

**Comment by poettering** [comment]
lgtm

**Comment by pothos** [comment]
Thank you for working on this! :tada:

**Pull request merged**